R/fairness_selection.R

```diff
@@ -0,0 +1,274 @@
+    }
+  }
+
+  if (is.null(q3)) {
+    q3 <- utils::menu(choices = c("Everything not positive", "Well-defined"), title = "Can the negative class be considered as everything not positive or is it well-defined?")
+  } else {
+    if (!(q3 %in% c(1, 2))) {
+      stop("Invalid input: The value of the third argument must be 1 (to indicate 'Everythig not positive') or 2 (to indicate 'Well-defined')")
+    }
+  }
+
+  if (is.null(q4)) {
+    q4 <- utils::menu(choices = c("False Positive", "False Negative", "No preference"), title = "What are the errors with the highest cost?")
+  } else {
+    if (!(q4 %in% c(1, 2, 3))) {
+      stop("Invalid input: The value of the fourth argument must be 1 (to indicate 'False Positive'), 2 (to indicate 'False Negative'), or 3 (to indicate 'No preference').")
+    }
+  }
+
+  if (q1 == 2) {
+    name <- "Disparate Impact"
+    measure <- "pp"
+  } else if (q1 == 1) {
+    if (q2 == 3) {
+      name <- "Equalized Odds"
+      measure <- "dp"
+    } else {
+      if (q2 == 1) {
+        if (q3 == 1) {
+          if (q4 == 1) {
+            name <- "Predictive Rate Parity"
+            measure <- "prp"
+          } else if (q4 == 2) {
+            name <- "Equal Opportunity"
+            measure <- "tprp"
+          } else if (q4 == 3) {
+            name <- "No measure with these characteristics is available in the Decision-Making Workflow"
+            measure <- "error"
+          } else {
+            stop("Invalid imput:  The value of the fourth argument must be 1 (to indicate 'False Positive'), 2 (to indicate 'False Negative'), or 3 (to indicate 'No preference').")
+          }
+        } else if (q3 == 2) {
+          if (q4 == 1) {
+            name <- "Specificity Parity"
+            measure <- "sp"
+          } else if (q4 == 2) {
+            name <- "Negative Predictive Rate Parity"
+            measure <- "npvp"
+          } else if (q4 == 3) {
+            name <- "Accuracy Parity"
+            measure <- "ap"
+          } else {
+            stop("Invalid imput:  The value of the fourth argument must be 1 (to indicate 'False Positive'), 2 (to indicate 'False Negative'), or 3 (to indicate 'No preference').")
+          }
+        } else {
+          stop("Invalid input: The value of the third argument must be 1 (to indicate 'Everythig not positive') or 2 (to indicate Well-defined')")
+        }
+      } else if (q2 == 2) {
+        if (q4 == 1) {
+          name <- "False Positive Rate Parity"
+          measure <- "fprp"
+        } else if (q4 == 2) {
+          name <- "False Negative Rate Parity"
+          measure <- "fnrp"
+        } else if (q4 == 3) {
+          name <- "No measure with these characteristics is available in the Decision-Making Workflow"
+          measure <- "error"
+        } else {
+          stop("Invalid imput:  The value of the fourth argument must be 1 (to indicate 'False Positive'), 2 (to indicate 'False Negative'), or 3 (to indicate 'No preference').")
```

Comment:
spelling error: `input` not `imput`. Also, use 1 space instead of 2 after `:`. Check for this in other strings as well.